File: frontend/src/app/features/wind-turbine-map/models/measurementDto.ts
export interface MeasurementDto {
  measurementId: number;
  turbineId: number;
  productId: string;
  turbineType: string;
  timestamp: string;

  // Dane pogodowe i fizyczne (Environment & Operational)
  windSpeedKmH: number;
  airTempCelsius: number;
  rpm: number;
  torque: number;
  powerGenerated: number;
  powerNetto: number;
  processTempCelsius: number;
  toolWear: number;

  // Flagi diagnostyczne
  deIcingActive: boolean;
  severity: 'HEALTHY' | 'CAUTION' | 'FAILURE' | string;
  targetLabel: string;
  predictionAI: string;
}

File: backend/src/main/java/pl/zimnya/wind_turbine_maintenance/turbine/dto/MeasurementDto.java
package pl.zimnya.wind_turbine_maintenance.turbine.dto;

import java.time.LocalDateTime;

public record MeasurementDto(
        Long measurementId,
        Long turbineId,
        String productId,
        String turbineType,
        LocalDateTime timestamp,

        // Dane pogodowe i fizyczne
        double windSpeedKmH,
        double airTempCelsius,
        int rpm,
        double torque,
        double powerGenerated,
        double powerNetto,
        double processTempCelsius,
        int toolWear,

        // Flagi diagnostyczne (używamy boolean)
        boolean deIcingActive,
        String severity,
        String targetLabel,
        String predictionAI
) {}
